bokehjs/src/coffee/models/tools/toolbar_template.tsx

```diff
@@ -0,0 +1,30 @@
+import * as DOM from "../../core/util/dom";
+
+interface ToolbarProps {
+  location: "above" | "below" | "left" | "right";
+  sticky: "sticky" | "non-sticky";
+  logo: "normal" | "grey" | null;
+}
+
+export default (props: ToolbarProps): HTMLElement => {
+  let logo;
+  if (props.logo != null) {
+    const cls = props.logo === "grey" ? "bk-grey" : null;
+    logo = <a href="http://bokeh.pydata.org/" target="_blank" class={["bk-logo", "bk-logo-small", cls]}></a>
+  }
+
+  return <div class={[`bk-toolbar-${props.location}`, `bk-toolbar-${props.sticky}`]}>
```

Comment:
I would prefer to have this:
```
  return
   <div ...>
      ...
   </div>
```
instead, but that won't work, because of semicolon inference in ES and thus TypeScript. The above code compiles to:
```
  return;
  <div ...>
    ...
  </div>
```
That's a lot of fun. At least TypeScript warns about unreachable code.